enhancements/observability/cluster-observability-operator-logging-stack.md

```diff
@@ -0,0 +1,374 @@
+
+When used with previous released CLO versions:
+1. COO creates a `Subscription` for CLO **without** enabling the Console plugin on the cluster-wide `Consoles` resource.
+2. COO creates a `ClusterLogging` resource setting `spec.visualization.type: console`.
+2. CLO creates and continues managing the [logging-view-plugin](logging-view-plugin) deployment resource.
+3. COO patches the `Consoles` resource when reconciling the `ObserbabilityUIPlugin` resource owned by `LoggingStack`.
+
+When used with future CLO version w/o [logging-view-plugin](logging-view-plugin) management:
+1. COO simply creates a `Subscription` for CLO to install the operator. A Console plugin option is omitted as not existing anymore.
+2. COO creates a `ClusterLogging` resource managing only `spec.collection` and `spec.logStorage`.
+3. COO creates a deployment for [logging-view-plugin](logging-view-plugin) when reconciling the `ObserbabilityUIPlugin` resource owned by `LoggingStack`.
+4. COO patches the `Consoles` resource when reconciling the `ObserbabilityUIPlugin` resource owned by `LoggingStack`.
+
+### Cluster Obserbability Operator design impact managing OLM-based operators
+
+When an operator (i.e. COO) managing the installation/upgrade of further operators (i.e. CLO and LO) as well as resources owned by the managed operators, they key consideration should be on when managed CRDs will registered on the Kube APIserver. The registration event is a cornerstone for the kubernetes clientsets used in COO to reconcile shared custom resources such as `ClusterLogging` and `Lokistack`. In particular it is mandatory for client abstration such as `informers` and `listers` to execute **only** on resource known by the Kube APIserver.
+
+Therefore the Cluster Observability Operator requires a major design overhaul that supports the following aspects:
+1. Independent reconciliation loops for installing/upgrading managed operators.
+2. Reconciliation loops for shared resources such as `ClusterLogging` and `Lokistack` only and only if registration of the corresponding CRD successful
+3. Robust signaling between the above two and in turn mindful setup and tear-down of underlying kubernetes clientsets/informers/listers.
+
+In summary the Cluster Observability Operator binary will hold two actively communicating sub-controllers each managing distinct aspects for an owned resource like `LoggingStack`, i.e.:
+1. **Operators-Controller**: This controller reconciles a `LoggingStack` resource into `OperatorGroup` and `Subscription` resources for CLO and LO only. It tracks success as well as failure conditions for each operator installation/upgrade phase on the `LoggingStack` status field.
```

Comment:
I think this makes sense and also informs the discussion above. Should a `LoggingStack` CRD really configure the OpGroup and Subscription.